singularity-eos/base/root-finding-1d/root_finding.hpp

```diff
@@ -228,22 +231,26 @@ template <typename T>
 PORTABLE_INLINE_FUNCTION Status findRoot(const T &f, const Real ytarget, Real xguess,
                                          const Real xmin, const Real xmax,
                                          const Real xtol, const Real ytol, Real &xroot,
-                                         const RootCounts &counts) {
+                                         const RootCounts *counts = nullptr) {
   Status status;
 
   // first check if we're at the max or min values
   const Real fmax = f(xmax);
   const Real errmax = fabs(fmax - ytarget) / (fabs(fmax) + ytol);
   if (errmax < ytol) {
     xroot = xmax;
-    counts.increment(0);
+    if (counts != nullptr) {
+      counts->increment(0);
```

Comment:
Why is `counts->increment()` called with `iteration_count` in `regula_falsi()` but with the `0` literal here?

Reply:
That line is inside a check for "we don't need to do any iterations because the guess is already correct." That's why the `if (errmax < ytol)`. It's before any iterations have been performed.